web/src/user_profile.js

```diff
@@ -94,6 +99,32 @@ export function update_user_profile_streams_list_for_users(user_ids) {
     }
 }
 
+export function update_bot_profile_data(bot_id) {
+    const bot = people.get_by_user_id(bot_id);
+    render_bot_profile_fields(bot);
+}
+
+export function update_user_profile_data(user, new_data) {
+    if (new_data.avatar_url !== undefined) {
+        $("#avatar").css("background-image", `url(${people.medium_avatar_url_for_person(user)})`);
+    }
+    if (new_data.delivery_email !== undefined) {
+        $("email").find(".value").text(new_data.delivery_email);
+    }
+    if (new_data.role !== undefined) {
+        $("#user-type").find(".value").text(new_data.role);
+    }
+    const user_type = {
+        name: user.full_name,
+        should_add_guest_user_indicator: people.should_add_guest_user_indicator(user.user_id),
+    };
+    $(".user_profile_name").html(render_user_full_name(user_type));
+}
```

Comment:
Updating a user's name is a bit tricky because when the `role` for a user is changed, there might be a need to modify the name by adding or removing the suffix `(guest)`. This depends on whether the user's role has been changed to or from a guest role.

Reply:
Can the name part also be in an if-condition like when name or role is changed?